elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -720,6 +975,63 @@ private static boolean containsDisallowedWords(String str, String splitter, Set<
                 : Arrays.stream(str.trim().toUpperCase(Locale.ENGLISH).split(splitter)).anyMatch(keywords::contains);
     }
 
+    private void validateArgumentExists(Table currentModel, Named currentColumn, String argName) {
+        // Arguments are not supported for Joins. Maintain a set of required Column arguments, and this set must be
+        // validated for each logical column which references this join.
+        if (currentColumn instanceof Join) {
+            ((Join) currentColumn).getRequiredColumnArgs().add(argName);
```

Comment:
A couple of problems with this:
1. We don't want to put internal validation logic into public contracts (Join configuration class).
2. validation functions should be idempotent as callers will expect them to be stateless.

We need to find another way to do this.  Maybe we need to parse all the join expressions and pull out their required arguments as step one.  At a later step, we can validate that any column which references the join also has all required arguments.  

We also want to make sure we are not duplicating work done by the SQL Reference Table (which does some validation work as well with respect to cycles).  SQL reference table is the only place today that understand the expansion of a join (joins can depend on other joins and so on).  I think we need to rethink how this validation logic responsibility is shared with the SQL Reference table.